src/openforms/forms/admin/form.py

```diff
@@ -113,17 +113,28 @@ class FormAdmin(
     search_fields = ("name", "internal_name")
 
     change_list_template = "admin/forms/form/change_list.html"
+    _global_config = None
+
+    @property
+    def global_config(self):
+        if not self._global_config:
+            self._global_config = GlobalConfiguration.get_solo()
+        return self._global_config
```

Comment:
ModelAdmin classes/instances are not thread safe, you should absolutely avoid sharing state on these instances.

The only places where it's acceptable if really needed is in methods that are called for every request, they typically have the `request` argument.